components/new_post/post_options.tsx

```diff
@@ -0,0 +1,279 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React, {ReactNode, useRef, useState} from 'react';
+
+import {FormattedMessage} from 'react-intl';
+
+import {Posts, Preferences} from 'mattermost-redux/constants/index';
+import {isPostEphemeral} from 'mattermost-redux/utils/post_utils';
+
+import {Locations} from 'utils/constants';
+import * as PostUtils from 'utils/post_utils';
+import {isMobile} from 'utils/utils';
+
+import {Post} from '@mattermost/types/posts';
+import {Emoji} from '@mattermost/types/emojis';
+
+import ActionsMenu from 'components/actions_menu';
+import DotMenu from 'components/dot_menu';
+import PostFlagIcon from 'components/post_view/post_flag_icon';
+import PostRecentReactions from 'components/post_view/post_recent_reactions';
+import PostReaction from 'components/post_view/post_reaction';
+import CommentIcon from 'components/common/comment_icon';
+
+type Props = {
+    post: Post;
+    teamId: string;
+    isFlagged: boolean;
+    removePost: (post: Post) => void;
+    enableEmojiPicker?: boolean;
+    isReadOnly?: boolean;
+    channelIsArchived?: boolean;
+    setActionsMenuInitialisationState: (initializationState: Record<string, boolean>) => void;
+    handleCommentClick?: (e: React.MouseEvent) => void;
+    handleJumpClick?: (e: React.MouseEvent) => void;
+    handleDropdownOpened?: (e: boolean) => void;
+    collapsedThreadsEnabled?: boolean;
+    shouldShowActionsMenu?: boolean;
+    showActionsMenuPulsatingDot?: boolean;
+    oneClickReactionsEnabled?: boolean;
+    recentEmojis: Emoji[];
+    isExpanded?: boolean;
+    hover?: boolean;
+    isMobileView: boolean;
+    a11yActive?: boolean;
+    hasReplies?: boolean;
+    isFirstReply?: boolean;
+    isSearchResultsItem?: boolean;
+    canReply?: boolean;
+    replyCount?: number;
+    location: 'CENTER' | 'RHS_ROOT' | 'RHS_COMMENT' | string;
```

Comment:
Are we sure this needs to accept an arbitrary string as an option?

Reply:
Oh, I see that we're passing `Locations.SEARCH` into this. In that case, we can replace these types with `keyof typeof Locations` to avoid repeating ourselves
```suggestion
    location: keyof typeof Locations;
```